src/coreclr/vm/instancecalli.cpp

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "common.h"
+
+void MethodDesc::GenerateFunctionPointerCall(DynamicResolver** resolver, COR_ILMETHOD_DECODER** methodILDecoder)
+{
+    STANDARD_VM_CONTRACT;
+    _ASSERTE(resolver != NULL);
+    _ASSERTE(methodILDecoder != NULL);
+    _ASSERTE(*resolver == NULL && *methodILDecoder == NULL);
+    _ASSERTE(IsIL());
+    _ASSERTE(GetRVA() != 0);
+
+    // Intrinsic must be a static method.
+    _ASSERTE(IsStatic());
+
+    MetaSig declarationSig(this);
+    UINT argCount = declarationSig.NumFixedArgs();
+
+    // Intrinsic must have at least the "this" argument and the IntPtr function pointer argument.
+    _ASSERTE(argCount >= 2);
+
+    SigTypeContext genericContext;
+    ILStubLinker sl(
+        this->GetModule(),
+        GetSignature(),
+        &genericContext,
+        this,
+        (ILStubLinkerFlags)ILSTUB_LINKER_FLAG_NONE);
+
+    ILCodeStream* pCode = sl.NewCodeStream(ILStubLinker::kDispatch);
+
+    // Copy the first arg which has the function pointer signature in order to add HASTHIS and EXPLICITTHIS.
```

Comment:
This comment states we are copying the first arg, but right below we call `SkipArg()`. That is a bit confusing. Perhaps reference the signature format from ECMA-335 we are following here. I do this a few times in the `UnsafeAccessor` logic so people know where to reference the format being assumed.

Reply:
The nature of those APIs are you have to do a SkipArg() + GetArgProps() to get a pointer to the first arg. GetArgProps() should be called "GetLastArg()" or something like that. Public access to the current parsing pointer is not provided.

I added the ECMA-335 link although we are treating the actual signature opaque in this method; just leaving it up to "skip()" semantics to get the start and end pointers so we can do a binary copy.